docs/design.md

```diff
@@ -0,0 +1,339 @@
+
+## <a name="job_service_interface">Job Service Interface
+```go
+// Start a new command.  Once started, the JobService will run the command concerntly, writing the output to a file
+// returns the command ID on success
+func (j *JobService) Start(ctx context.Context, cert *x509.Certificate, command string, args []string) (string, error)
+
+// Stop a running job.  Returns nil error on success or error 
+func (j *JobService) Stop(ctx context.Context, commandID string) error
+
+// Query a command.  On success a QueryResponse struct is returned
+func (j *JobService) Query(ctx context.Context, commandId string) (*QueryResponse, error)
+
+type QueryResponse struct {
+	Id string        //command ID
+	State string     // state of the command: running, canceled, error or completed
+	Command string   // the command which was run
+	Args []string    // arguments to the command
+}
+
+// Returns a struct to read command output from the output file (See <a name="#command_output">Command Output</a> for more details
+func (j *JobService) GetOutput(ctx context.Context, string commandId) (*CommandOutput, error)
+
+
+```
+
+
+## <a name="command_output"></a>Command Output
+The commandOutput struct is responsible for streaming the contents of the command files, which contain the output of commands run by the job service. The commandOutput is returned by the `GetOutput` method in the JobService and consists of:
+```go
+type commandOutput struct {
+	Id string           //command ID
+	File string         // path to the command output file
+        Out chan string     // channel the file's content is written to
+	Repo JobRepo        // Job Repository object
+}
+
+// ReadData does the following:
+// 1) open the command file
+// 2) read the contents from the file and write it to the Out channel
+// 3) when the end of the file is reached, check status of the job in the JobRepo.
+//     a) if the job is not running, exit
+//     b) if the job is still running:
+//          i) sleep for a short period of time
+//          ii) check if there is more data to read.
+//        This loop will continue until the job stops and the JobService updates the JobRepo status to anything other than running
+//
+// The idea is that the caller can start ReadData as a go function, read the results in the Out chan and stream the results back to the gRPC layer.
+func (c *commandOutput) ReadData(ctx context.Context) error
+
+```
+# <a name="_dp6btk223uz"></a>Jobs Repository
+The job repository preserves the state of each job. For this project, the jobs repository will be stored in memory.
+
+The schema is:
+
+```
+     command_id        | user cert    |     state    |   command     |    args       |    output_file.  | exit        | error

```

Comment:
This does not seem correct:
`state: command which was run`

Reply:
fixed